language-support/ts/daml-react/index.test.ts

```diff
@@ -307,236 +307,257 @@ describe('useFetchByKey', () => {
+      expect(mockStreamFetchByKeys).toHaveBeenLastCalledWith(Foo, [key1]);
+      expect(result.current.fetchResult).toEqual(found(contract));
+
+      mockStreamFetchByKeys.mockClear();
+      mockStreamFetchByKeys.mockReturnValueOnce(stream);
+      act(() => result.current.setKey(key2));
+      expect(mockStreamFetchByKeys).toHaveBeenCalledTimes(1);
+      expect(mockStreamFetchByKeys).toHaveBeenLastCalledWith(Foo, [key2]);
+    });
 
-  describe('createLedgerContext', () => {
-    test('contexts can nest', () => {
-      const innerLedger = createLedgerContext();
-      const innerTOKEN = "inner_TOKEN";
-      const innerPARTY = "inner_PARTY";
-      const outerLedger = createLedgerContext('Outer');
-      const outerTOKEN = "outer_TOKEN";
-      const outerPARTY = "outer_PARTY";
+    describe('createLedgerContext', () => {
+      test('contexts can nest', () => {
+        const innerLedger = createLedgerContext();
+        const innerTOKEN = "inner_TOKEN";
+        const innerPARTY = "inner_PARTY";
+        const outerLedger = createLedgerContext('Outer');
+        const outerTOKEN = "outer_TOKEN";
+        const outerPARTY = "outer_PARTY";
 
-      const innerWrapper: ComponentType = ({children}) => React.createElement(innerLedger.DamlLedger, {token:innerTOKEN, party:innerPARTY}, children);
-      const r1 = renderHook(() => innerLedger.useParty(), {wrapper:innerWrapper});
+        const innerWrapper: ComponentType = ({children}) => React.createElement(innerLedger.DamlLedger, {token:innerTOKEN, party:innerPARTY}, children);
+        const r1 = renderHook(() => innerLedger.useParty(), {wrapper:innerWrapper});
 
-      expect( r1.result.current).toBe(innerPARTY);
+        expect( r1.result.current).toBe(innerPARTY);
 
-      const outerWrapper: ComponentType = ({children}) => React.createElement(outerLedger.DamlLedger, {token:outerTOKEN, party:outerPARTY}, innerWrapper({children}) );
-      const r2 = renderHook(() => outerLedger.useParty(), {wrapper:outerWrapper});
-      expect( r2.result.current).toBe(outerPARTY);
+        const outerWrapper: ComponentType = ({children}) => React.createElement(outerLedger.DamlLedger, {token:outerTOKEN, party:outerPARTY}, innerWrapper({children}) );
+        const r2 = renderHook(() => outerLedger.useParty(), {wrapper:outerWrapper});
+        expect( r2.result.current).toBe(outerPARTY);
 
-      const r3 = renderHook(() => innerLedger.useParty(), {wrapper:outerWrapper});
-      expect( r3.result.current).toBe(innerPARTY);
+        const r3 = renderHook(() => innerLedger.useParty(), {wrapper:outerWrapper});
+        expect( r3.result.current).toBe(innerPARTY);
 
-    })
+      });
+    });
+  };
+}
 
-  })
-});
+describe(
+  'useStreamFetchByKey',
+  streamk(
+    useStreamFetchByKey,
+    (k: string): (() => string) => (): string => k,
+    {loading: true, contract: null},
+    (c: object | null): {loading: boolean; contract: object | null} => ({loading: false, contract: c})));
+describe(
```

Comment:
Would a test be useful that actually tests multiple queries or keys or is this somewhat redundant in this mocked setup?

Reply:
Given that we're completely mocking the entire stream stack here, I don't think it would add any value at all to test at this level. I'm otherwise happy with the current test coverage of the underlying `@daml/ledger` functions, but would be open to try and add more of that in a separate PR if you're not.

Reply:
I agree with you, just wanted to check if I had all the context.